MonstieWash/Assets/Scripts/Eraser.cs

```diff
@@ -0,0 +1,233 @@
+{
+    [SerializeField] private Tool tool;
+
+    private List<Erasable> m_erasables = new();
+    private Vector2Int m_mouseScreenPos = Vector2Int.zero;
+    private Vector2Int m_prevMouseScreenPos = Vector2Int.zero;
+
+    private struct Erasable
+    {
+        public Erasable(GameObject obj, Sprite sprite, byte[] maskPixels)
+        {
+            this.obj = obj;
+            this.sprite = sprite;
+            this.maskPixels = maskPixels;
+        }
+
+        public GameObject obj;
+        public Sprite sprite;
+        public byte[] maskPixels;
+
+        public void ApplyMask()
+        {
+            var colors = sprite.texture.GetPixels();
+            var newColors = new Color[maskPixels.Length];
+
+            for (int i = 0; i < newColors.Length; i++)
+            {
+                newColors[i].r = colors[i].r;
+                newColors[i].g = colors[i].g;
+                newColors[i].b = colors[i].b;
+                newColors[i].a = Mathf.Min((255 - maskPixels[i])/255f, colors[i].a);
+            }
+
+            sprite.texture.SetPixels(newColors, 0);
+            sprite.texture.Apply(false);
+        }
+    }
+
+    private void Awake()
+    {
+        InitializeTool();
+        PopulateErasables();
+    }
+
+    private void Update()
+    {
+        transform.localScale = Vector3.one * tool.size;
+
+        if (!Input.GetMouseButton(0)) return;
+        if (!MouseMoved()) return;
+
+        foreach (var erasable in m_erasables)
+        {
+            if (UpdateErasableMask(erasable)) erasable.ApplyMask();
+        }
+    }
+
+    /// <summary>
+    /// Sets up the tool and related variables ready for use.
+    /// </summary>
+    private void InitializeTool()
+    {
+        tool.Initialize();
+        transform.localScale = Vector3.one * tool.size;
+        GetComponent<SpriteRenderer>().sprite = tool.mask;
+    }
+
+    /// <summary>
+    /// Populates the erasables list, setting up the sprites ready for drawing.
+    /// </summary>
+    private void PopulateErasables()
+    {
+        var tempErasables = GameObject.FindGameObjectsWithTag("Erasable");
+
+        foreach (var erasable in tempErasables)
+        {
+            var spriteRenderer = erasable.GetComponent<SpriteRenderer>();
+            spriteRenderer.sprite = DuplicateSprite(spriteRenderer.sprite);
+            var sprite = spriteRenderer.sprite;
+
+            var newErasable = new Erasable(erasable, sprite, new byte[sprite.texture.width * sprite.texture.height]);
+            m_erasables.Add(newErasable);
+        }
+    }
+
+    /// <summary>
+    /// Duplicates the given sprite.
+    /// </summary>
+    /// <param name="sprite">The sprite to duplicate.</param>
+    /// <returns>The duplicated sprite.</returns>
+    private Sprite DuplicateSprite(Sprite sprite)
+    {
+        var textureCopy = new Texture2D(sprite.texture.width, sprite.texture.height, sprite.texture.format, false);
+        textureCopy.alphaIsTransparency = true;
```

Comment:
Texture2D.alphaIsTransparency is editor only and can't be changed unless done in TextureImporter.alphaIsTransparency. Also it breaks exporting builds

Reply:
Hmm that's v unfortunate. I'll have a look at another way to do this that is legal.